homework/java/shelest_vladimir/login_server/src/test/java/by/jrr/user/UserTest.java

```diff
@@ -0,0 +1,89 @@
+package by.jrr.user;
+
+import by.jrr.bean.User;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Objects;
+
+public class UserTest {
+    @Test
+    public void toString_Test() {
+        User user = new User("Login", "Password");
+        String expected = "User{" +
+                "login = '" + "Login" + '\'' +
+                ", password = '" + "Password" + '\'' +
+                ", attempt = " + "3" +
+                ", is blocked = " + "false" +
+                '}';
+        String actual = user.toString();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equals_Test() {
+        User user = new User("Login", "Password");
+        User anotherUser = new User("Login", "AnotherPassword");
+
+        boolean expected = true;
+        boolean actual = user.equals(anotherUser);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equals_AnotherUser_Test() {
+        User user = new User("Login", "Password");
+        User anotherUser = new User("AnotherLogin", "Password");
+
+        boolean expected = false;
+        boolean actual = user.equals(anotherUser);
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void equalsWithSameUser_Test() {
+        User user = new User("Login", "Password");
+
+        boolean expected = true;
+        boolean actual = user.equals(user);
```

Comment:
Our static analysis engine detects that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-field-with-itself-sa-field-self-comparison).
This method compares a field with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.